### PR TITLE
Send registry auth token for service deploy

### PR DESCRIPTION
**Pull request by nishanttotla**
Follow up to #23584 

This PR passes the registry auth token for `docker service deploy`.

cc @tiborvass @tonistiigi @aaronlehmann - it's not clear what the best way to set and retrieve the `--registry-auth` flag in this case.

### CODE REVIEW

**Comment by tiborvass** [comment]
LGTM

File: api/client/stack/deploy.go [hunk 1]
```diff
@@ -181,24 +181,42 @@ func deployServices(
 			cspec.User = *service.User
 		}
 
+		// TODO(nishanttotla): retrieve sendAuth from a flag
+		sendAuth := true
```
Comment (nishanttotla) [inline api/client/stack/deploy.go]: Currently `sendAuth` is just just set to true, but must be set based on the CLI flag. It's not clear how to retrieve the flag in this case.

Comment (aaronlehmann) [inline api/client/stack/deploy.go]: I suggest adding an argument to `deployServices` with the value of the flag. `namespace` is also passed through from a flag:

`return deployServices(ctx, dockerCli, bundle.Services, opts.namespace)`

**Comment by aaronlehmann** [comment]
ping @nishanttotla what's the status on this one?

**Comment by nishanttotla** [comment]
@aaronlehmann will fix it today, sorry got sidetracked with some other PRs.

**Comment by nishanttotla** [comment]
@aaronlehmann @tiborvass I updated the PR to use a `--registry-auth` flag. PTAL

**Comment by stevvooe** [comment]
LGTM

**Comment by aaronlehmann** [comment]
LGTM

Looks like the `registry-auth` flag needs to be documented through.

**Comment by nishanttotla** [comment]
@aaronlehmann you're right, I was preparing a follow-up PR for that. Is that fine, or should I add docs changes to this one?

**Comment by tiborvass** [comment]
@nishanttotla preferrably this same PR

**Comment by nishanttotla** [comment]
Updated docs. Please let me know if I need to update in more places!

**Comment by aaronlehmann** [comment]
Please update `docs/reference/commandline/deploy.md`.

**Comment by nishanttotla** [comment]
@aaronlehmann updated. Sorry I hadn't rebased and seems like this file was added in the meantime.

**Comment by aaronlehmann** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**